Integrations/QRadar/QRadar.py

```diff
@@ -792,31 +795,34 @@ def transform_single_asset_to_hr(asset):
 
 
 def create_single_asset_result_and_enrich_endpoint_dict(asset, endpoint_dict, full_values):
-    asset_dict = {'ID': asset['id']}
-    for interface in asset['interfaces']:
-        if full_values:
-            endpoint_dict['MACAddress'].append(interface['mac_address'])
-        for ip_address in interface['ip_addresses']:
-            endpoint_dict['IPAddress'].append(ip_address['value'])
+    asset_dict = {'ID': asset.get('id')}
+    if 'interfaces' in asset:
+        for interface in asset['interfaces']:
+            if full_values:
+                endpoint_dict['MACAddress'].append(interface['mac_address'])
+            for ip_address in interface['ip_addresses']:
+                endpoint_dict['IPAddress'].append(ip_address['value'])
     if full_values:
-        domain_name = get_domain_name(asset['domain_id'])
-        endpoint_dict['Domain'].append(domain_name)
+        if 'domain_id' in asset:
+            domain_name = get_domain_name(asset['domain_id'])
+            endpoint_dict['Domain'].append(domain_name)
     # Adding values found in properties of the asset
     enrich_dict_using_asset_properties(asset, asset_dict, endpoint_dict, full_values)
     return asset_dict
 
 
 def enrich_dict_using_asset_properties(asset, asset_dict, endpoint_dict, full_values):
-    for prop in asset['properties']:
-        if prop['name'] in ASSET_PROPERTIES_NAMES_MAP:
-            asset_dict[ASSET_PROPERTIES_NAMES_MAP[prop['name']]] = {'Value': prop['value'],
-                                                                    'LastUser': prop['last_reported_by']}
-        elif prop['name'] in ASSET_PROPERTIES_ENDPOINT_NAMES_MAP:
-            endpoint_dict[ASSET_PROPERTIES_ENDPOINT_NAMES_MAP[prop['name']]] = prop['value']
-        elif full_values:
-            if prop['name'] in FULL_ASSET_PROPERTIES_NAMES_MAP:
-                asset_dict[FULL_ASSET_PROPERTIES_NAMES_MAP[prop['name']]] = {'Value': prop['value'],
-                                                                             'LastUser': prop['last_reported_by']}
+    if 'properties' in asset:
+        for prop in asset['properties']:
+            if prop['name'] in ASSET_PROPERTIES_NAMES_MAP:
```

Comment:
there are also direct key access in these conditions as well, use `.get()` instead